compiler/rustc_mir_build/src/build/matches/test.rs

```diff
@@ -153,21 +154,9 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         test: &Test<'tcx>,
         make_target_blocks: impl FnOnce(&mut Self) -> Vec<BasicBlock>,
     ) {
-        let place: Place<'tcx>;
-        if let Ok(test_place_builder) =
-            place_builder.try_upvars_resolved(self.tcx, self.typeck_results)
-        {
-            place = test_place_builder.into_place(self.tcx, self.typeck_results);
-        } else {
-            return;
```

Comment:
`into_place` ICEs if `try_upvars_resolved` would have returned `Err`. So this early return was always dead code as I could not figure out when it would ever happen and nothing in our test suite actually hit it.